website/frontend/src/components/pages/State/State.tsx

```diff
@@ -76,6 +57,27 @@ const State: VFC = () => {
     [api, programId]
   );
 
+  const handleBackButtonClick = () => {
+    navigate(-1);
+  };
+
+  const handleSubmit = (values: FormValues) => {
+    const payload = preparePayload(values.payload);
+
+    readState(payload);
+  };
+
+  const typeStructures = useMemo(() => {
+    if (types && stateInput) {
+      const decodedTypes = decodeHexTypes(types);
+
+      return {
+        manual: createPayloadTypeStructure(stateInput, decodedTypes, true),
+        payload: createPayloadTypeStructure(stateInput, decodedTypes),
+      };
+    }
+  }, [types, stateInput]);
```

Comment:
This part is the same on the few pages, so it can be standalone `getPayloadTypeStructures` function